### ISSUE TITLE
Using SHAPs on unseen data to understand model's predictions

### ISSUE THREAD

**Issue by ETTAN93** [unit8co/darts#2571]
Assuming I have a model that is initialized as such:

```
model_estimator = LightGBMModel(
    lags=None,
    lags_past_covariates=[-3,-2,-1].
    lags_future_covariates=[-3,-2,-1],
    output_chunk_length=3
)
```
I then create the ShapExplainer object by fitting it to the training set.
```
shap_explain = ShapExplainer(model_estimator)
explanations = shap_explain_train.summary_plot()
```
Assuming now I want to use the ShapExplainer to explain data in the test set, what should be defined as the foreground series?
I tried not providing the foreground series since target lags is 0 but it seems like that is not possible. In that case, should the foreground series, past_covariates and future covariates be the same series passed to the `model.predict` function, i.e. the foreground_series should end at the first prediction timestamp t=0.

```
shap_explain.explain(
    foreground_series = local_lgbm_hf_output.hf_data_dict['target_hf'][: test_set_start_date],
    foreground_past_covariates = local_lgbm_hf_output.hf_data_dict['past_cov_hf'][test_set_start_date: end_date],
    foreground_future_covariates = local_lgbm_hf_output.hf_data_dict['future_cov_hf'][test_set_start_date: end_date],
    horizons = [3]
)
```

Or should the foreground_series start from the test_set_start_date?
```
shap_explain.explain(
    foreground_series = local_lgbm_hf_output.hf_data_dict['target_hf'][test_set_start_date: ],
    foreground_past_covariates = local_lgbm_hf_output.hf_data_dict['past_cov_hf'][test_set_start_date: end_date],
    foreground_future_covariates = local_lgbm_hf_output.hf_data_dict['future_cov_hf'][test_set_start_date: end_date],
    horizons = [3]
)
```
